File: project_euler/Problem015/Problem015.js
function Problem015() {
    let target = 20
    let border = target << 1
    let result = 1
    for (let i = border - 1; i > target; i -= 2) {
        result = Math.floor(result * 2 * i / ((border - i + 1) >>> 1))
    }
    console.log(result)
}

module.exports = {
    Problem015 : Problem015
};

File: project_euler/Problem015/Problem015.cpp
#include <iostream>
using namespace std;

class Problem015 {
public:
    void solution() {
        int target = 20;
        int border = target << 1;
        long long result = 1L;
        for (int i = border - 1; i > target; i -= 2) {
            result = (result << 1) * i / ((border - i + 1) >> 1);
        }
        cout << result << endl;
    }
};
